src/flamenco/vm/fd_vm.c

```diff
@@ -393,12 +393,15 @@ fd_vm_init(
   vm->trace = trace;
   vm->sha = sha;
 
-  // TODO: call fd_vm_setup_state_for_execution here once it is trace-aware
+  /* Unpack the configuration */
+  int err = fd_vm_setup_state_for_execution( vm );
+  if( FD_UNLIKELY( err != FD_VM_SUCCESS ) ) {
+    return NULL;
+  }
```

Comment:
We need to also switch to use `fd_vm_init` everywhere, otherwise this will break the ledger tests. `fd_vm_setup_state_for_execution` is currently not being called by the bpf loader.